test/test_from_bag.py

```diff
@@ -0,0 +1,136 @@
+
+	def test_vector3_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/vector3']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = Vector3()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_point_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/point']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = Point()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_quaternion_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/quaternion']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = Quaternion()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_transform_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/transform']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = Transform()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_pose_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/pose']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = Pose()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_compressed_image_from_bag(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/compressedimage']):
+			arr = ros_numpy.numpify(bagmsg)
+			msg = CompressedImage()
+			for slot in msg.__slots__:
+				# Go thru all the slots, data, header...
+				setattr(msg, slot, getattr(bagmsg, slot))
+			arr2 = ros_numpy.numpify(msg)
+			np.testing.assert_equal(arr, arr2)
+
+	def test_pose_from_bag_with_different_md5(self):
+		bag = Bag('test.bag')
+		# There is only one message
+		for topic, bagmsg, time in bag.read_messages(topics=['/pose']):
+			# use a different md5 than the installed one, for reference
+			# ros-indigo-geometry-msgs 1.11.9-0trusty-20160627-170848-0700
+			# Pose msg has md5sum e45d45a5a1ce597b249e23fb30fc871f
+			# so we use one made up from:
+			from different_md5_Pose import Pose as different_md5_Pose
```

Comment:
@eric-wieser  If you are wondering why I added a __init__.py file to the test folder, is so I can import this custom Pose message definition with a different md5sum. (_md5sum is a read-only attribute so I didn't find another way of testing with a different md5sum).

Reply:
This won't work on python 3 unless you change it to `from .different_md5_Pose import Pose`.

Instead of bundling that file, could you just generate a test message as part of the test cmake section?

Reply:
Done on the from import.

I'm very unfamiliar with cmake... and it's just a test, I can't really work on it right now.